charts/newrelic-k8s-metrics-adapter/templates/_helpers.tpl

```diff
@@ -1,11 +1,17 @@
 {{/* vim: set filetype=mustache: */}}
 
 {{- /* Allow to change pod defaults dynamically based if we are running in privileged mode or not */ -}}
-{{- define "common.securityContext.podDefaults" -}}
+{{- define "newrelic-k8s-metrics-adapter.securityContext.pod" -}}
+{{- if include "newrelic.common.securityContext.pod" . -}}
+{{- include "newrelic.common.securityContext.pod" . -}}
+{{- else -}}
 fsGroup: 1001
 runAsUser: 1001
 runAsGroup: 1001
 {{- end -}}
+{{- end -}}
+
```

Comment:
Probably I'm missing something, but I'm asking just in case I'm not: wouldn't it work to set those values as defaults in `values.yaml` and directly use `newrelic.common.securityContext.pod` in `deployment.yaml`?

Reply:
The thing here is that if you put something in the `values.yaml` directly it is used always and any kind of `.global.podSecurityContext` is ignored always.

So we honor what it comes from `global`, then `local` and in case the user did not provide any value for security pod's context, then we use our defaults.